src/storage/src/render/upsert.rs

```diff
@@ -462,9 +462,12 @@ where
                 }
             }
 
+            let batch_size = events.len();
+
             match state
                 .merge_snapshot_chunk(
                     events.drain(..),
+                    batch_size,
```

Comment:
Was not sure how to get the size otherwise, lmk if there's a better way.

Reply:
I think what would be better is:
- if the size of the batch we send it is < 1 / 2 the capacity of the buffer, we can shrink the buffer, instead of passing in `batch_size` here. `batch_size` is just the maximum size

Reply:
Makes sense, I will update. But I would still need to send the `batch_size` though? The type is an intoiterator in the `merge_snapshot_chunk` method, so no way to get the length without iterating.

Reply:
(`batch_size ` is the length of `events` getting passed to this method)